DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Centre/Configuration/RegistrationPromptsControllerTests.cs

```diff
@@ -433,18 +503,43 @@ public void RegistrationPromptBulkPost_updates_temp_data_and_redirects_to_edit()
             // Given
             var inputViewModel = new BulkRegistrationPromptAnswersViewModel("Test\r\nAnswer", false, 1);
             var initialEditViewModel = new EditRegistrationPromptViewModel(1, "Prompt", false, "Test");
-            var expectedViewModel = new EditRegistrationPromptViewModel(1, "Prompt", false, "Test\r\nAnswer");
-            var initialTempData = new EditRegistrationPromptData(initialEditViewModel);
-
-            registrationPromptsController.TempData.Set(initialTempData);
+            var initialTempData = new EditRegistrationPromptData
+            {
+                PromptNumber = initialEditViewModel.PromptNumber,
+                Prompt = initialEditViewModel.Prompt,
+                Mandatory = initialEditViewModel.Mandatory,
+                OptionsString = initialEditViewModel.OptionsString,
+                Answer = initialEditViewModel.Answer,
+                IncludeAnswersTableCaption = initialEditViewModel.IncludeAnswersTableCaption,
+            };
+            A.CallTo(
+                () => multiPageFormDataService.GetMultiPageFormData<EditRegistrationPromptData>(
+                    MultiPageFormDataFeature.EditRegistrationPrompt,
+                    registrationPromptsController.TempData
+                )
+            ).Returns(initialTempData);
+            
 
             // When
             var result = registrationPromptsController.EditRegistrationPromptBulkPost(inputViewModel);
 
             // Then
             using (new AssertionScope())
             {
-                AssertEditTempDataIsExpected(expectedViewModel);
+                A.CallTo(
+                    () => multiPageFormDataService.SetMultiPageFormData(
+                        A<EditRegistrationPromptData>.That.Matches(
+                            d => d.PromptNumber == initialTempData.PromptNumber &&

```

Comment:
This assertion is present twice in the file. Extract to another `AssertMultiPageFormDataIsUpdatedCorrectly` function?

Reply:
They aren't quite the same assertions, the types of the objects we're checking against aren't the same.